server/world/mcdb/provider.go

```diff
@@ -141,6 +144,76 @@ func (p *Provider) SaveSettings(s *world.Settings) {
 	p.saveDifficulty(s.Difficulty)
 }
 
+// LoadPlayerSpawnPosition loads the players spawn position stored in the level.dat from their UUID.
+func (p *Provider) LoadPlayerSpawnPosition(uuid uuid.UUID) (pos mgl64.Vec3, err error) {
```

Comment:
Maybe it'd be better to return a bool alongside the error, to specify if the player spawn didn't exist or if there's an actual error in the data.

Reply:
`(pos mgl64.Vec3, exists bool, err error)`